ConcurrentModificationException with --repos from the cli.

**Describe the bug**
When you use `--repos mavencentral` in the command line, you get a `ConcurrentModificationException`

> **Note**
> Our use case is not use `mavencentral` but our company mirror of maven central. But for the reproducer I thought it is easier to use `mavencentral`, because everybody has it.

**To Reproduce**
Steps to reproduce the behavior:

Create a small jbang script `parser.java`:

```java
//usr/bin/env jbang "$0" "$@" ; exit $?

//DEPS com.alibaba:fastjson:1.2.68
//DEPS org.yaml:snakeyaml:1.26

import java.util.Arrays;
import java.util.List;

import com.alibaba.fastjson.JSON;
import org.yaml.snakeyaml.Yaml;

public class parser {

    public static void main(String... args) {
    	List<String> value = Arrays.asList("a", "b");
    	
    	String jsonString = JSON.toJSONString(value);
    	System.out.println("-- JSON --");
    	System.out.println(jsonString);
    	
        Yaml yaml = new Yaml();
        String output = yaml.dump(value);
        System.out.println("-- YAML --");
        System.out.println(output);
    }
}
```

Run it with:
```
 jbang --fresh --repos mavencentral parser.java 
```

Usage of `--fresh` make the error appear all the time.
Without `--fresh` it depends (and I could not yet figure out when it isn't working)

jbang is failing with:
```
[jbang] [ERROR] public class java.util.ConcurrentModificationException
```
(see the verbose output at the bottom)

**Expected behavior**
It should work.

**JBang version**

* `0.93.0` 🟢 --> works
* `0.94.0` 🔴 --> first version where it is broken
* `0.95.0` 🔴 --> still broken with this version

**Additional context**
Verbose output:
```
[jbang] jbang version 0.95.0
[jbang] Building as fresh build explicitly requested.
[jbang] System Java version detected as 8
[jbang] System Java version matches requested version 8
[jbang] Repositories: mavencentral=https://repo1.maven.org/maven2/
[jbang] Resolving dependencies...
[jbang] com.alibaba:fastjson:jar:1.2.68
         org.yaml:snakeyaml:jar:1.26
Done
[jbang] Dependencies resolved
[jbang] Building jar...
[jbang] compile: /home/work/.sdkman/candidates/java/current/bin/javac -classpath /home/work/.m2/repository/com/alibaba/fastjson/1.2.68/fastjson-1.2.68.jar:/home/work/.m2/repository/org/yaml/snakeyaml/1.26/snakeyaml-1.26.jar -d /home/work/.jbang/cache/jars/parser.java.481b4230d20182ba53d86c7a39ed4b3df738648eb45768b891ede46fdcbe6bb9.jar.tmp /home/work/tmp/jbang/parser.java
[jbang] Deleting folder /home/work/.jbang/cache/jars/parser.java.481b4230d20182ba53d86c7a39ed4b3df738648eb45768b891ede46fdcbe6bb9.jar.tmp
[jbang] Repositories: mavencentral=https://repo1.maven.org/maven2/, mavencentral=https://repo1.maven.org/maven2/
[jbang] Resolving dependencies...
[jbang] [ERROR] public class java.util.ConcurrentModificationException
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)
	at java.util.ArrayList$Itr.next(ArrayList.java:859)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.addRemoteRepo(MavenWorkingSessionImpl.java:315)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:71)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:36)
	at dev.jbang.dependencies.MavenRepo.apply(MavenRepo.java:40)
	at dev.jbang.dependencies.DependencyUtil.lambda$resolveDependenciesViaAether$2(DependencyUtil.java:161)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:161)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:131)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:77)
	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:67)
	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:377)
	at dev.jbang.source.generators.JarCmdGenerator.generateCommandLineList(JarCmdGenerator.java:48)
	at dev.jbang.source.generators.BaseCmdGenerator.generate(BaseCmdGenerator.java:36)
	at dev.jbang.cli.Run.doCall(Run.java:104)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:136)
	at dev.jbang.cli.BaseCommand.call(BaseCommand.java:20)
	at picocli.CommandLine.executeUserObject(CommandLine.java:1953)
	at picocli.CommandLine.access$1300(CommandLine.java:145)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2352)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2346)
	at dev.jbang.cli.JBang$3.handle(JBang.java:145)
	at dev.jbang.cli.JBang$3.handle(JBang.java:140)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2179)
	at picocli.CommandLine.execute(CommandLine.java:2078)
	at dev.jbang.Main.main(Main.java:14)
```

I think the main issue is that the same repo is added twice in the `[jbang] Repositories` line.

This is similar to this issue https://github.com/jbangdev/jbang/issues/1293

What I now did is to add the same test twice:

* https://github.com/jmini/jbang/commit/b9f9814 -> on top of `main`
* https://github.com/jmini/jbang/commit/e9ec6fb -> on top of the tag `v0.94.0`

Running the command:
```
./gradle test --tests *TestRun.testReposWorksWithFresh
```

In the second case (on top of `0.94.0`), it is failing with:
```

> Task :test

TestRun > testReposWorksWithFresh() FAILED
    java.util.ConcurrentModificationException at TestRun.java:1979

1 test completed, 1 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
```

And inside the linked test report, the stacktrace looks similar (failing during `MavenWorkingSessionImpl.addRemoteRepo`):
```
java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:909)
	at java.util.ArrayList$Itr.next(ArrayList.java:859)
	at org.jboss.shrinkwrap.resolver.impl.maven.MavenWorkingSessionImpl.addRemoteRepo(MavenWorkingSessionImpl.java:315)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:71)
	at org.jboss.shrinkwrap.resolver.impl.maven.ConfigurableMavenResolverSystemImpl.withRemoteRepo(ConfigurableMavenResolverSystemImpl.java:36)
	at dev.jbang.dependencies.MavenRepo.apply(MavenRepo.java:40)
	at dev.jbang.dependencies.DependencyUtil.lambda$resolveDependenciesViaAether$2(DependencyUtil.java:161)
	at java.util.ArrayList.forEach(ArrayList.java:1257)
	at dev.jbang.dependencies.DependencyUtil.resolveDependenciesViaAether(DependencyUtil.java:161)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:131)
	at dev.jbang.dependencies.DependencyUtil.resolveDependencies(DependencyUtil.java:77)
	at dev.jbang.dependencies.DependencyResolver.resolve(DependencyResolver.java:67)
	at dev.jbang.source.RunContext.resolveClassPath(RunContext.java:363)
	at dev.jbang.source.generators.JarCmdGenerator.generateCommandLineList(JarCmdGenerator.java:48)
	at dev.jbang.source.generators.BaseCmdGenerator.generate(BaseCmdGenerator.java:35)
	at dev.jbang.cli.TestRun.testReposWorksWithFresh(TestRun.java:1979)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:688)
```

---

@quintesse I don't know the jbang code base well enough to know if my test would have an interest or if this a duplication with something you already are checking somewhere else. Important would be to detect a regression in this area next time.

I can open a PR to add the test, otherwise I think we can close this issue since it seems to be fixed with the next version.

--- a/src/test/java/dev/jbang/cli/TestRun.java
+++ b/src/test/java/dev/jbang/cli/TestRun.java
@@ -2065,4 +2065,42 @@ public class TestRun extends BaseTest {
 			run.doCall();
 		});
 	}
+
+	@Test
+	void testReposWorksWithFresh() throws IOException {
+		File f = Util.getCwd().resolve("classpath_example.java").toFile();
+
+		String content = ""
+				+ "///usr/bin/env jbang \"$0\" \"$@\" ; exit $?\n"
+				+ "\n"
+				+ "//DEPS log4j:log4j:1.2.17\n"
+				+ "\n"
+				+ "import org.apache.log4j.Logger;\n"
+				+ "import org.apache.log4j.BasicConfigurator;\n"
+				+ "\n"
+				+ "import java.util.Arrays;\n"
+				+ "\n"
+				+ "class classpath_example {\n"
+				+ "\n"
+				+ "	static final Logger logger = Logger.getLogger(classpath_example.class);\n"
+				+ "\n"
+				+ "	public static void main(String[] args) {\n"
+				+ "		BasicConfigurator.configure();\n"
+				+ "		logger.info(\"Welcome to jbang\");\n"
+				+ "	}\n"
+				+ "}";
+		Util.writeString(f.toPath(), content);
+
+		CommandLine.ParseResult pr = JBang	.getCommandLine()
+											.parseArgs("run", "--fresh", "--repos", "mavencentral", f.getPath());
+		Run run = (Run) pr.subcommand().commandSpec().userObject();
+
+		RunContext ctx = run.getRunContext();
+		Code code = ctx.forResource(f.getPath());
+		code = run.prepareArtifacts(code, ctx);
+
+		String result = code.cmdGenerator(ctx).generate();
+
+		assertThat(result, containsString("log4j-1.2.17.jar"));
+	}
 }

